src/VisualStudio/CSharp/Impl/Debugging/CSharpProximityExpressionsService.cs

```diff
@@ -71,13 +73,23 @@ internal partial class CSharpProximityExpressionsService : IProximityExpressions
             return true;
         }
 
+        /// <summary>
+        /// Returns null indicating a failure.
+        /// </summary>
         public async Task<IList<string>> GetProximityExpressionsAsync(
             Document document,
             int position,
             CancellationToken cancellationToken)
         {
-            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            return Do(tree, position, cancellationToken);
+            try
+            {
+                var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                return Do(tree, position, cancellationToken);
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
```

Comment:
Ditto: why are we changing the behavior of the data API here? Put this at the VS shim interop?